Libraries/Animated/src/Animated.js

```diff
@@ -16,12 +16,22 @@ var AnimatedImplementation = require('AnimatedImplementation');
 var Image = require('Image');
 var Text = require('Text');
 var View = require('View');
-var ScrollView = require('ScrollView');
 
-module.exports = {
-  ...AnimatedImplementation,
+let AnimatedScrollView;
+
+const Animated = {
   View: AnimatedImplementation.createAnimatedComponent(View),
   Text: AnimatedImplementation.createAnimatedComponent(Text),
   Image: AnimatedImplementation.createAnimatedComponent(Image),
-  ScrollView: AnimatedImplementation.createAnimatedComponent(ScrollView),
+  get ScrollView() {
+    // Make this lazy to avoid circular reference.
+    if (!AnimatedScrollView) {
+      AnimatedScrollView = AnimatedImplementation.createAnimatedComponent(require('ScrollView'));
+    }
+    return AnimatedScrollView;
+  },
 };
+
+Object.assign((Animated: Object), AnimatedImplementation);
```

Comment:
@sahrens Using `{ ...AnimatedImplementation, ... }` will compile to `Object.assign({}, AnimatedImplementation, ...)` but there seems to be an issue with getters on iOS, the getter doesn't get assigned but it's value is instead, losing the getter altogether.

This is why I used `Object.assign(Animated, AnimatedImplementation)` instead to avoid having to copy `Animated` props and getters get preserved properly. However doing that causes a bunch of flow issues because we are assigning props that don't exist on the initial object.